tool/teleport/common/teleport.go

```diff
@@ -484,6 +509,260 @@ func OnStart(clf config.CommandLineFlags, config *service.Config) error {
 	return service.Run(context.TODO(), *config, nil)
 }
 
+// GenerateKeys generates TLS and SSH keypairs.
+func GenerateKeys() (private, sshpub, tlspub []byte, err error) {
+	privateKey, publicKey, err := native.GenerateKeyPair()
+	if err != nil {
+		return nil, nil, nil, trace.Wrap(err)
+	}
+
+	sshPrivateKey, err := ssh.ParseRawPrivateKey(privateKey)
+	if err != nil {
+		return nil, nil, nil, trace.Wrap(err)
+	}
+
+	tlsPublicKey, err := tlsca.MarshalPublicKeyFromPrivateKeyPEM(sshPrivateKey)
+	if err != nil {
+		return nil, nil, nil, trace.Wrap(err)
+	}
+
+	return privateKey, publicKey, tlsPublicKey, nil
+}
+
+func authenticatedUserClientFromIdentity(ctx context.Context, fips bool, proxy utils.NetAddr, id *auth.Identity) (auth.ClientI, error) {
+	tlsConfig, err := id.TLSConfig(nil /* cipherSuites */)
```

Comment:
nit: i think we suppose to use this for FIPS
https://github.com/gravitational/teleport/blob/73cc189b0e9636d418c4470ecce0d9af5dae2f02/lib/defaults/defaults.go#L680-L681
may want to update RegisterParams as well